src/material/snack-bar/testing/snack-bar-harness.ts

```diff
@@ -74,13 +74,21 @@ export class MatSnackBarHarness extends ComponentHarness {
     return (await this._simpleSnackBarMessage()).text();
   }
 
+  /** Gets whether the snack-bar has been dismissed. */
+  async isDismissed(): Promise<boolean> {
+    // We consider the snackbar dismissed if it's not in the DOM. We can assert that the
+    // element isn't in the DOM by seeing that its width and height are zero.
+    const dimensions = await (await this.host()).getDimensions();
+    return dimensions.height === 0 && dimensions.width === 0;
```

Comment:
I'm not sure how this would behave if it was called mid-animation, right after `dismiss` was called. Does the test harness guarantee that all the animations have settled?

Reply:
Technically the element is already gone, and could not be loaded by the `HarnessLoader` again. Though consumers still have a reference to it, so it might be good to expose a method to `TestElement`? 

The method could then use protractor `isPresent`, and in testbed, check if a parent element exists.

My thinking initially was that folks just use the loader to check if dismissed or not. Though I agree that this is not convenient.

Reply:
@devversion I don't follow what you mean- while the exit animation plays, the snackbar container would still be in the DOM, no?

Seems like the easiest thing to do for the animation would be to add a marker attribute to an existing snackbar and check for that as well. 

As for the logic of checking whether the element is in the DOM: checking the geometry gets the job done for now, I would rather just use that until we need to add an `isInDom` method for some other use case.

Reply:
@jelbourn Sorry. I was not chatting about the mid-animation thing @crisbeto mentioned. I was basically saying that figuring out whether an element is in the DOM is a general use-case for harnesses and that checking the geometry is a symptom of the problem that consumers can have references to harnesses while the actual host element is already gone.

I agree that checking the geometry does the job for now, but I think long-term adding a method for `isPresent` might be useful in `TestElement`.

Reply:
@crisbeto added handling for the animation, PTAL